tally_counter_block.py

```diff
@@ -0,0 +1,36 @@
+from nio.signal.base import Signal
+from nio.properties import VersionProperty
+from .counter_block import Counter
+
+
+class TallyCounter(Counter):
+
+    """ A block that counts the number of signals
+    that are processed by it.
+
+    Enriches incoming signals with `tally`, including a key for each
+    unique evaluation of `group_by` since the last `reset()`. The value of 
+    each key is the `cumulative_count` for that `group`.
+
+    """
+    version = VersionProperty("0.0.1")
+
+    def process_group(self, signals, key):
+        """ Executed on each group of incoming signal objects.
+        Increments the appropriate count and generates an informative
+        output signal.
+
+        """
+        count = self._get_count_from_signals(signals)
+
+        self.logger.debug(
+            "Ready to process {} signals in group {}".format(count, key)
+        )
+        self._cumulative_count[key] = self._cumulative_count.get(key, 0)
+        self._cumulative_count[key] += count
+        enriched_signal = self.get_output_signal({
+            "tally": self._cumulative_count,
+            "count": count,
+            "group": key
+        }, signals[0])
```

Comment:
Should we enrich all of the incoming signals? Or just the first one?

Reply:
While copying this out of the counter block I had the same question!
https://github.com/nio-blocks/counter/blob/master/counter_block.py#L175

Basically it boils down to the fact that block is going to turn a list of _n_  signals into a list of 1 signal, so in the case you have more than one signal for `group` in a list you have to pick one to enrich.

Reply:
Oh yeah, duh. Ok cool that makes sense - I figured there was a good reason.